ev3sim/attach.py

```diff
@@ -44,11 +44,14 @@ def main(passed_args = None):
     print_builtin = print
     def print_mock(*objects, sep=' ', end='\n', **kwargs):
         message = sep.join(str(obj) for obj in objects) + end
-        source = shared_data['thread_ids'][threading.get_ident()]
-        shared_data['actions_queue'].put(('send_log', (message, source)))
-        if not args.send_logs:
-            print_builtin(message, end='', **kwargs)
-
+        thread_id = threading.get_ident()
+        if thread_id in shared_data['thread_ids']:
+            source = shared_data['thread_ids'][thread_id]
+            shared_data['actions_queue'].put(('send_log', (message, source)))
+            if not args.send_logs:
+                print_builtin(message, end='', **kwargs)
+        else:
+            print_builtin(*objects, sep=sep, end=end, **kwargs)
```

Comment:
We might be able to avoid having a nested if loop here:
```suggestion
        thread_id = threading.get_ident()
        if thread_id in shared_data['thread_ids'] and args.send_logs:
            source = shared_data['thread_ids'][thread_id]
            shared_data['actions_queue'].put(('send_log', (message, source)))
        else:
            print_builtin(*objects, sep=sep, end=end, **kwargs)
```

Reply:
The data should be sent over regardless of `args.send_logs` - remember this flag only changes where the printing occurs (This means later on we'd be saving the output to a file on the sim thread regardless)

lmk if I'm misunderstanding, otherwise feel free to merge whenever you're happy.

Reply:
Oh right, I forgot that we were sending the logs even if we didn't want them to be printed in the simulator process, sorry. Ignore this then 🙂